### PR TITLE
remove kube-rbac-proxy as a dependency

### PR DESCRIPTION
**Pull request by sadlerap**
We need to set up a ServiceMonitor for our service.  We could use kube-rbac-proxy, but that has apparently been deprecated by kubebuilder, and they recommend moving away from it.  Instead, we can do the authentication and authorization checks inside our metrics server.

To do this, we need to set up the ServiceMonitor with a ServiceAccount with permissions to talk to `/metrics` from our metrics service.

### CODE REVIEW

File: operator/config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -12,7 +12,6 @@ resources:
 # Comment the following 4 lines if you want to disable
 # the auth proxy (https://github.com/brancz/kube-rbac-proxy)
 # which protects your /metrics endpoint.
-# - auth_proxy_service.yaml
-# - auth_proxy_role.yaml
-# - auth_proxy_role_binding.yaml
-# - auth_proxy_client_clusterrole.yaml
+- auth_proxy_role.yaml
+- auth_proxy_role_binding.yaml
+- auth_proxy_client_clusterrole.yaml
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply operator/config/rbac/kustomization.yaml:17]: nope, we need them this time.

- `auth_proxy_role.yaml` and `auth_proxy_role_binding.yaml` give our operator permissions to create subject access reviews and token reviews, both of which are needed by the `filters.WithAuthenticationAndAuthorization` callback.
- `auth_proxy_client_clusterrole.yaml` is a role bound to the `metrics-reader` service account by the `metrics-reader` ClusterRoleBinding (defined in `operator/config/metrics/service-account.yaml`).  It allows the service account to access our service's `/metrics` endpoint.

File: operator/config/metrics/monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+spec:
+  endpoints:
+    - interval: 15s
+      scheme: http
+      path: /metrics
+      port: metrics
+      authorization:
+        credentials:
+          key: token
+          name: metrics-reader
```
Comment (dperaza4dustbit) [inline operator/config/metrics/monitor.yaml:23]: the service account creation automatically creates a token secret which name is not predictable. How are you going to make sure there is a secret named metrics-reader?

**Review by dperaza4dustbit — CHANGES_REQUESTED** [review]

File: operator/config/metrics/monitor.yaml (hunk 2 above)
Comment (sadlerap) [reply operator/config/metrics/monitor.yaml:23]: The secret gets generated in `operator/config/metrics/kustomization.yaml`.

Comment (sadlerap) [reply operator/config/metrics/monitor.yaml:23]: We also write the name of this generated secret into the service monitor via a replacement in `operator/config/default/kustomization.yaml`:

```yaml
replacements:
- source:
    fieldPath: metadata.name
    kind: Secret
    name: metrics-reader
  targets:
  - fieldPaths:
    - spec.endpoints.*.authorization.credentials.name
    select:
      group: monitoring.coreos.com
      kind: ServiceMonitor
      name: metrics-proxy
```

**Review by dperaza4dustbit — APPROVED** [review]
/lgtm

**Pull request merged**